manim/mobject/text/code_mobject.py

```diff
@@ -89,549 +64,186 @@ def construct(self):
-        if self.file_name:
-            self._ensure_valid_file()
-            self.code_string = self.file_path.read_text(encoding="utf-8")
-        elif code:
-            self.code_string = code
-        else:
-            raise ValueError(
-                "Neither a code file nor a code string have been specified.",
-            )
-        if isinstance(self.style, str):
-            self.style = self.style.lower()
-        self._gen_html_string()
-        strati = self.html_string.find("background:")
-        self.background_color = self.html_string[strati + 12 : strati + 19]
-        self._gen_code_json()
+        if code_file is None and code_string is None:
+            raise ValueError("Either a code file or a code string must be specified.")
 
-        self.code = self._gen_colored_lines()
-        if self.insert_line_no:
-            self.line_numbers = self._gen_line_numbers()
-            self.line_numbers.next_to(self.code, direction=LEFT, buff=self.line_no_buff)
-        if self.background == "rectangle":
-            if self.insert_line_no:
-                foreground = VGroup(self.code, self.line_numbers)
-            else:
-                foreground = self.code
-            rect = SurroundingRectangle(
-                foreground,
-                buff=self.margin,
-                color=self.background_color,
-                fill_color=self.background_color,
-                stroke_width=self.background_stroke_width,
-                stroke_color=self.background_stroke_color,
-                fill_opacity=1,
-            )
-            rect.round_corners(self.corner_radius)
-            self.background_mobject = rect
+        if code_file is not None:
+            code_file = Path(code_file)
+            code_string = code_file.read_text(encoding="utf-8")
+            lexer = guess_lexer_for_filename(code_file.name, code_string)
         else:
-            if self.insert_line_no:
-                foreground = VGroup(self.code, self.line_numbers)
+            if language is not None:
+                lexer = get_lexer_by_name(language)
             else:
-                foreground = self.code
-            height = foreground.height + 0.1 * 3 + 2 * self.margin
-            width = foreground.width + 0.1 * 3 + 2 * self.margin
+                lexer = guess_lexer(code_string)
 
-            rect = RoundedRectangle(
-                corner_radius=self.corner_radius,
-                height=height,
-                width=width,
-                stroke_width=self.background_stroke_width,
-                stroke_color=self.background_stroke_color,
-                color=self.background_color,
-                fill_opacity=1,
+        formatter = HtmlFormatter(
+            style=formatter_style,
+            noclasses=True,
+            cssclasses="",
+        )
+        soup = BeautifulSoup(
+            highlight(code_string, lexer, formatter), features="html.parser"
+        )
+        self._code_html = soup.find("pre")
```

Comment:
In general, `BeautifulSoup.find()` returns `Tag | NavigableString | None`. Only `Tag` has a `children` attribute. To make MyPy happy, we could add an assertion:

```suggestion
        self._code_html = soup.find("pre")
        assert isinstance(self._code_html, Tag)
```

Reply:
LGTM!